src/lib/storage/index/base_index.hpp

```diff
@@ -38,6 +38,19 @@ class BaseIndex : private Noncopyable {
   // For now we use an iterator over a vector of chunkoffsets as the GroupKeyIndex works like this
   using Iterator = std::vector<ChunkOffset>::const_iterator;
 
+  /**
+   * Predicts the memory consumption in bytes of creating an index with the specific index implementation <type>
+   * on a Chunk with the following statistics:
+   *
+   * row_count - overall number of rows
+   * value_count - number of distinct values
+   * value_bytes - (average) size of a single value in bytes
+   *
+   * If no prediction is possible, this shall fail.
```

Comment:
Is this intended for the other not yet supported index types? There, a prediction would be possible, but is simply not implemented. In all other cases, a prediction should be possible, right?

Reply:
Yes. I changed the comment to "If no prediction is possible (or it is not implemented yet), this shall fail" to reflect that.